src/task_userdefined.md

```diff
@@ -0,0 +1,148 @@
+# User-defined Custom Tasks in RMF Task
+
+>Note: User-defined custom tasks are currently experimental
+
+
+When dealing with rmf_task, there are two packages. 
+- rmf_task 
+- rmf_task_sequence
+
+`rmf_task` provides APIs and base classes for defining and managing Tasks in RMF. A Task is defined as an object that generates phases which are a meaningful sequence of steps that results in a desirable outcome.  Each task will have a description and a component that allows us to model the state of the robot after completing the task given its initial state and also a component that will command the actual robot to perform the task.
+
+`rmf_task_sequence` provides an out of the box implementation of rmf_task where a Task object is defined by a sequence of phases. The phases that such tasks generate will thus match the sequence of phases used to define them. 
+The phases defined in rmf_task_sequence are in turn a collection of events which also have components to model the end state and command the robot during the event. Presently the following events are supported  are defined [here](https://github.com/open-rmf/rmf_task/tree/main/rmf_task_sequence/include/rmf_task_sequence/events)
+Users can then construct arbitrary definitions of tasks by stringing together a sequence of such phases/events. RMF is capable of planning and executing such tasks.
+
+Users will be working on `rmf_fleet_adapter` layer. In this layer API is restricted to use only `rmf_task_sequence` to perform tasks and only certain events are supported their description can be found [here](https://github.com/open-rmf/rmf_ros2/tree/main/rmf_fleet_adapter/schemas)
+
+The `rmf_fleet_adapter` layer acts as an API that users can use and it supports customised behaviour on only `perform_action` in already existing  events mentioned [here](https://github.com/open-rmf/rmf_ros2/tree/main/rmf_fleet_adapter/schemas) .
+Users can only add custom tasks in `perform_action`. RMF passes the command to the platform-specific end of the fleet adapter integration and formally releases control of the bot until the action is finished.
+To use a custom task in `perform_action` users need to use two parts of the API. 
+1. [FleetUpdateHandle::add_performable_action](https://github.com/open-rmf/rmf_ros2/blob/8440488d5583edc5a5b7226326aa2a8d41dad975/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/FleetUpdateHandle.hpp#L188-L201)
+This consists of two things one is the `category` of the action and the second is the `consider` based on which it would be decided whether to accept the action or not.
```

Comment:
```suggestion
This consists of two parts: the first one is the `category` of the action and the second is the `consider` part, based on which it would be decided whether to accept the action or not.
```